### PR TITLE
C++ API parity: at::Tensor::is_leaf

### PR DESCRIPTION
**Pull request by pbelevich**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #26332 C++ API parity: at::Tensor::requires_grad_
* #26251 C++ API parity: at::Tensor::detach
* #26217 C++ API parity: at::Tensor::version
* #26216 C++ API parity: at::Tensor::output_nr
* **#26186 C++ API parity: at::Tensor::is_leaf**

Differential Revision: [D17427580](https://our.internmc.facebook.com/intern/diff/D17427580)

### CODE REVIEW

**Review by yf225 — APPROVED** [review]
Thanks a lot @pbelevich !
